packages/nodejs/src/probes/index.ts

```diff
@@ -4,7 +4,72 @@ import { Probes } from "../interfaces"
 /**
  * The Minutely probes object.
  */
-export class BaseProbes extends EventEmitter implements Probes {
+export class BaseProbes implements Probes {
+  #probes: InternalProbes
+  #stopped = false
+
+  constructor({ stopped = false } = {}) {
+    this.#probes = new StartedProbes()
+    if (stopped) this.stop()
+  }
+
+  public stop(): this {
+    this.#probes.clear()
+    this.#probes = new StoppedProbes()
+    this.#stopped = true
+    return this
+  }
+
+  get isStopped(): boolean {
+    return this.#stopped
+  }
+
+  get count(): number {
+    return this.#probes.count
+  }
+
+  public register(name: string, fn: () => void): this {
+    this.#probes.register(name, fn)
+    return this
+  }
+
+  public unregister(name: string): this {
+    this.#probes.unregister(name)
+    return this
+  }
+
+  public clear(): this {
+    this.#probes.clear()
+    return this
+  }
+}
+
+type InternalProbes = {
```

Comment:
Right! It's kind of a mouthful, but it's definitely better than "Internal".

Curious: how would you feel about `ProbeRunner` and `NoopProbeRunner`/`StoppedProbeRunner`? This might be just me, but I wouldn't call this a "collection". I expect something named "collection" to be a container, usually a class that wraps over a list or map primitive, sometimes with some helper methods specific to the purpose of the container -- but something that doesn't _do anything_ with the elements other than contain them and provide you ways to access them.

Thanks for all the comments on the naming, by the way. It's all over the place in this PR.

Reply:
Ah yes these things also start the probes and is not just a collection. The runner suffix would work.

Otherwise you need to move the runner part to the BaseProbes and add an if-statement to see if it's running now or not... No, that's gets more complicated. That would be a thing we need to do if we also want a manual "start" function, which we decided to leave out for now.

Reply:
Went with `ProbeRunner` for the interface and `BaseProbeRunner`/`NoopProbeRunner` for the classes.